src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -0,0 +1,138 @@
+from pyspiffe.workloadapi.workload_api_client import WorkloadApiClient
+from pyspiffe.internal.defaults import Defaults
+
+class DefaultWorkloadApiClient(WorkloadApiClient):
+    """
+    Default implementation for a SPIFFE Workload API Client.
+    """
+
+    # TODO: Not sure we need all the arguments, see newClient()
+    def __init__(self, spiffe_socket_path=None, exponential_backoff_policy=None,
+                 executor_service=None):
+
+        if spiffe_socket_path == None:
+            self.spiffe_socket_path=Defaults.spiffe_socket_path
+        else:
+            self.spiffe_socket_path=spiffe_socket_path
+
+        # TODO defaults for these, or delete them entirely
+        self.exponential_backoff_policy=exponential_backoff_policy
+        self.executor_service=executor_service
+
+
+    # TODO: Is this pythonic? Any benefit of this being a factory object vs
+    # just using the constructor?
+    @classmethod
+    def new_client(cls, spiffe_socket_path=None, exponential_backoff_policy=None,
+                  executor_service=None):
+        """
+        Create a new Workload API Client.
+
+        Parameters:
+
+        spiffeSocketPath (String) [optional]: Path to Workload API UDS.
+        exponentialBackoffPolicy (?? Type): Custom backoff policy instance. Do we need?
+        executorService (?? Type): Custom gRPC executor. Do we need?
+
+        Returns:
+
+        DefaultWorkloadApiClient: New Workload API Client object.
+        """
+
+        return cls(spiffe_socket_path=spiffe_socket_path,
+                   exponential_backoff_policy=exponential_backoff_policy,
+                   executor_service=executor_service)
+
+
+    def fetch_x509_context(self):
+        """
+        Fetches an X.509 context on a one-shot blocking call.
+
+        Fetch an instance of an X509Context containing the X.509
+        materials fetched from the Workload API,
+
+        Returns:
+
+        X509Context: The X.509 context.
+        """
+
+        pass
+
+
+    def watch_x509_context(self, watcher):
+        """
+        Watches for X.509 context updates.
+
+        A new Stream to the Workload API is opened for each call to
+        this method, so that the client starts getting updates
+        immediately after the Stream is ready and doesn't have to wait
+        until the Workload API dispatches the next update based on the
+        SVIDs TTL.
+
+        Parameters:
+
+        watcher (?? type): Watcher callback object.
+        """
+
+        pass
+
+
+    def fetch_jwt_svid(self, audiences, subject=None):
+        """
+        Fetches a SPIFFE JWT-SVID on one-shot blocking call.
+
+        Parameters:
+
+        audiences (set of String): Set of audiences for the JWT.
+        subject (String) [Optional]: SPIFFE ID Subject for the JWT.
+
+        Returns:
+
+        JwtSvid: Instance of JwtSvid object
```

Comment:
nit: I'm not sure about the _best_ name in python for `JwtSvid`, as it is a _JWT-SVID_

Reply:
Me neither! Open to suggestions...

Reply:
I think that according to PEP8 the idiomatic way is to have acronyms in all-caps, but `JWTSVID` isn't probably easier to read. I'd say let's use `JwtSvid`.

Reference: https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles
`Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.`